### PR TITLE
Add tests for post_type_s4b

### PR DESCRIPTION
**Pull request by kosgrz**
#### Summary
This PR adds some tests for the post_type_s4b components and boilerplate code required to run these tests.

### CODE REVIEW

File: webapp/babel.config.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        ['@babel/preset-react', {
+            useBuiltIns: true,
+        }],
+    ],
+    plugins: [
+        '@babel/plugin-proposal-class-properties',
+        '@babel/plugin-syntax-dynamic-import',
+        '@babel/proposal-object-rest-spread',
+        '@babel/plugin-proposal-optional-chaining',
+    ],
```
Comment (hanzei) [inline webapp/babel.config.js:28]: Should the template file from https://github.com/mattermost/mattermost-plugin-starter-template/blob/master/webapp/babel.config.js be used here?

File: webapp/tests/i18n_mock.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (hanzei) [inline webapp/tests/i18n_mock.json:1]: Is this file needed?

File: webapp/package.json [hunk 3]
```diff
@@ -8,11 +8,14 @@
   "scripts": {
     "build": "webpack --mode=production",
     "lint": "eslint --ignore-pattern node_modules --ignore-pattern dist --ext .js --ext .jsx . --quiet",
-    "fix": "eslint --ignore-pattern node_modules --ignore-pattern dist --ext .js --ext .jsx . --quiet --fix"
+    "fix": "eslint --ignore-pattern node_modules --ignore-pattern dist --ext .js --ext .jsx . --quiet --fix",
+    "test": "jest --forceExit --detectOpenHandles --verbose",
+    "test:watch": "jest --watch",
+    "test-ci": "jest --forceExit --detectOpenHandles --maxWorkers=2"
```
Comment (hanzei) [inline webapp/package.json:14]: I'm 1/5 that the template from https://github.com/mattermost/mattermost-plugin-starter-template/blob/master/webapp/package.json should be used, given that you are already touching the file.

File: webapp/babel.config.js (hunk 1 above)
Comment (kosgrz) [reply webapp/babel.config.js:28]: It's almost the same. I've just removed typescript features as they're not used in this plugin.

File: webapp/package.json (hunk 3 above)
Comment (kosgrz) [reply webapp/package.json:14]: The template uses additional scripts like `build:watch`, `debug` and `debug:watch`, that are used by the Makefile there, but not in this plugin. Although these features can be very helpful, I'm 4/5 that we should add them in a separate PR, as they aren't strictly related to tests.
Also, I removed `.ts` and `.tsx` options from these scripts, because we don't have TypeScript files in this plugin.

File: webapp/tests/i18n_mock.json (hunk 2 above)
Comment (kosgrz) [reply webapp/tests/i18n_mock.json]: You're right. This file is not needed. I removed it in the latest commit :+1:

File: webapp/babel.config.js (hunk 1 above)
Comment (mickmister) [reply webapp/babel.config.js:28]: @kosgrz Do you see value in introducing Typescript to the plugin? The config we use allows for a gradual migration. For example, you can turn any file in the project to Typescript and it will compile properly with Javascript files importing it or vise versa.
